include/boost/uuid/name_generator.hpp

```diff
@@ -1,21 +1,27 @@
 // Boost name_generator.hpp header file  ----------------------------------------------//
 
 // Copyright 2010 Andy Tompkins.
+// Copyright 2017 James E. King, III
+
 // Distributed under the Boost Software License, Version 1.0. (See
 // accompanying file LICENSE_1_0.txt or copy at
-// http://www.boost.org/LICENSE_1_0.txt)
+//  http://www.boost.org/LICENSE_1_0.txt)
 
 #ifndef BOOST_UUID_NAME_GENERATOR_HPP
 #define BOOST_UUID_NAME_GENERATOR_HPP
 
-#include <boost/uuid/uuid.hpp>
-#include <boost/uuid/sha1.hpp>
+#include <boost/concept_check.hpp>
+#include <boost/cstdint.hpp>
 #include <boost/static_assert.hpp>
+#include <boost/uuid/detail/md5.hpp>
```

Comment:
Please, avoid including unnecessary code. I would suggest moving `basic_name_generator` to a separate (new) header and leaving only `name_generator` in name_generator.hpp. The other typedefs with their respective includes can go in their new headers.

Another way to do this is create a name_generator directory with the headers, each header defining one component, and making name_generator.hpp include all headers from that directory. The headers in the name_generator directory can be named after the components they define.

The point is to allow users to include only the bits they need and avoid the rest. No one will use the MD5-based generator, so this is dead code 99% of the time and only slows down compilation and adds potential problems with compilation.